api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -77,6 +78,21 @@ public interface Scan<ThisT, T extends ScanTask, G extends ScanTaskGroup<T>> {
    */
   ThisT includeColumnStats();
 
+  /**
+   * Create a new scan from this that loads the column stats for the specific columns with each data
+   * file. If the columns set is empty or <code>null</code> then all column stats will be kept, if
+   * {@link #includeColumnStats()} is set.
+   *
+   * <p>Column stats include: value count, null value count, lower bounds, and upper bounds.
+   *
+   * @param columnsToKeepStats column ids from the table's schema
+   * @return a new scan based on this that loads column stats for specific columns.
+   */
+  default ThisT columnsToKeepStats(Set<Integer> columnsToKeepStats) {
```

Comment:
I think this should be `keepStatsFor` to align with the readability conventions.

Reply:
Why not simply overload `includeColumnStats()` with a version that accepts a collection of columns? I feel that will be easier to interpret as folks are already familiar with `includeColumnStats()` that currently loads stats for all columns.

Reply:
Done